#165399662 use es6 features through out  the app

NawasNaziru opened the pull request:
What does this PR do?
  Have es6 features like modules and arrow functions used through out.

Description of task to be completed?
  Replace old javascript module.exports and functions with es6 modules and arrow functions.

How should this be manually tested?
  Re-run the application testfile and test the API endpoints with nodemon to make sure nothing breaks.

Any background context you want to provide?
There is the need for use of latest language features to make future integration with tools easy and smooth.

What are the relevant pivotal tracker stories?
#165399662

NawasNaziru committed:
feat(everywhere): use es6 features
 - ensure use of es6 modules
[Delivers #165399662]

NawasNaziru committed:
ch(code refactor): use-es6-features
  -ensure every Javascript file uses the es6 features
 [Delivers #165399662]

phemmelliot commented on server/app.js:
```diff
@@ -1,121 +1,130 @@
+
+app.set('port', process.env.PORT || 3000);
+
+var server = app.listen(app.get('port'), () => {
```
var is an es5 keyword, change it to es6

NawasNaziru replied:
Thanks for the observation. I didn't notice it.

phemmelliot commented on .eslintrc.json:
```diff
@@ -0,0 +1,5 @@
+    "rules": {
+        "no-var": off
+    }
+}
```
Add a new line to all files

NawasNaziru replied:
Do you mean new line after every single line of code?

phemmelliot commented on UI/scripts/main.js:
```diff
@@ -4,17 +4,17 @@
 /* eslint-disable func-names */
```
Remove the eslint disable lines

NawasNaziru replied:
I have done that in a separate branch by addding the .eslintrc file you recommended. But do I add the file to .gitignore?

phemmelliot commented on server/app.js:
```diff
@@ -1,121 +1,130 @@
+import Debug from 'debug';
+const debug = Debug('Express4');
+
+import  {routesApi} from './routes/index';
```
Instead of using destructured import or export you can use default export and just import this without the destructuring

NawasNaziru replied:
Ok! Done.

phemmelliot commented on server/app.js:
```diff
@@ -1,121 +1,130 @@
+ console.log('Express server listening on localhost:3000');
+});
+
+export {app};
```
Make this a default export and a new blank line to this file as well

NawasNaziru replied:
New blank line on top of the file?

phemmelliot commented on server/controllers/accounts.js:
```diff
@@ -156,3 +156,5 @@ module.exports.delete = function (req, res) {
   }
 };
+const Num_of_handlers = 3;
+export default Num_of_handlers;
```
Add  a new line here

NawasNaziru replied:
Ok

phemmelliot commented on server/routes/index.js:
```diff
@@ -0,0 +1,27 @@
+// transaction routes
+router.post('/transactions/:accountNumber/debit', ctrlTxn.debit);
+router.post('/transactions/:accountNumber/credit', ctrlTxn.credit);
+export {router as routesApi};
```
Make this a default export and add a new line

NawasNaziru replied:
Ok

NawasNaziru committed:
ch(all files):implement all corrections in the PR raised from this branch.
 -use .eslintrc file
 -Add new line at the bottom of all files
 -use default import and export without destructuring
 -use let or const instead of var
[Delivers #165399662]

NawasNaziru committed:
ch(testing):use-es6-features
 - Mocha test file was fixed after broken as a result of using es6 module syntax
[Delivers #165399662]
